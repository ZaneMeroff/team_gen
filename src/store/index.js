import Vue from "vue"
import Vuex from "vuex"

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    teamNum: 2,
    playerNum: 2,
    playerList: ["", ""],
  },

  mutations: {

    plusteamNum: (state) => state.teamNum++,
    minusteamNum: (state) => state.teamNum !== 2 && state.teamNum--,

    plusplayerNum: (state) => state.playerNum++,
    minusplayerNum: (state) => state.playerNum !== 2 && state.playerNum--,

    updatePlayerList: (state, data) => state.playerList = data,
    updatePlayerName: (state, data) => state.playerList[data.id] = data.name,
  },

  actions: {},
  modules: {},
})
