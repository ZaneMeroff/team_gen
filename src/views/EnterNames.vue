<template>
  <div>

    <router-link class="router-link" to="/">
      <ArrowBtn :arrowDirection="'left'"/>
    </router-link>

    <router-link class="router-link" to="/displayResults">
      <ArrowBtn :arrowDirection="'right'"/>
    </router-link>

    <NameInput
      v-for="(player, index) in playerNum"
      :id="index"
    />

  </div>
</template>

<script>
  // need spec with snapshot

  import ArrowBtn from "../components/ArrowBtn"
  import NameInput from "../components/NameInput"

  export default {
    name: "EnterNames",
    components: {
      ArrowBtn,
      NameInput,
    },
    methods: {},
    computed: {

      playerNum() {
        return this.$store.state.playerNum
      },
    },
  }
</script>

<style scoped>

</style>
