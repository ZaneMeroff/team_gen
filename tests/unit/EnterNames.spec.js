import { createLocalVue, shallowMount } from '@vue/test-utils'
import store from "@/store/index"
import Vuex from "vuex"
import VueRouter from "vue-router"
import EnterNames from "@/views/EnterNames"

// --------- Vuex & Router setup ------------

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(VueRouter)

// ------------------------------------------

describe("EnterNames", () => {

  describe("snapshots", () => {

    it("should render with default html", () => {
      const component = shallowMount(EnterNames, { store, localVue })

      expect(component.element).toMatchSnapshot()
    })

    it("should render error message if showErr is true", () => {
      const component = shallowMount(EnterNames, { store, localVue })

      expect(component.element).toMatchSnapshot()
    })
  })

  describe("data", () => {

    describe("showErr", () => {

      it("should return a default value", () => {
        const component = shallowMount(EnterNames, { store, localVue })

        expect(component.vm.$data.showErr).toEqual(false)
      })
    })
  })

  describe("methods", () => {

    describe("validateInputs", () => {

      it("should call evaluateError with true if playerList contains an empty string", () => {
        const component = shallowMount(EnterNames, { store, localVue })
        const spy = jest.spyOn(component.vm, "evaluateError")

        component.vm.validateInputs()

        expect(spy).toHaveBeenCalledWith(true)
      })

      it("should call evaluateError with false if playerList doesn't contain an empty string", () => {
        const component = shallowMount(EnterNames, { store, localVue })
        component.vm.evaluateError = jest.fn()
        // ****************************************
        // ****************************************
        //     need to refactor with spy
        // ****************************************
        // ****************************************
        store.state.playerList = [ "Bob", "Joe" ]
        component.vm.validateInputs()

        expect(component.vm.evaluateError).toHaveBeenCalledWith(false)
        store.state.playerList = ["", ""]
      })
    })

    describe("evaluateError", () => {

      it.skip("should route to displayResults if error is false", () => {
        // ****************************************
        // ****************************************
        //     failing at this.$router.push()
        // ****************************************
        // ****************************************
        const component = shallowMount(EnterNames, { store, localVue })

        component.vm.evaluateError(false)

        expect(component.vm.$router).toEqual(["/displayResults"])
      })

      it("should set showErr to true if error is true", () => {
        const component = shallowMount(EnterNames, { store, localVue })

        expect(component.vm.$data.showErr).toEqual(false)

        component.vm.evaluateError(true)

        expect(component.vm.$data.showErr).toEqual(true)
      })
    })
  })

  describe("computed properties", () => {

    describe("playerList", () => {

      it("should return default playerList", () => {
        const component = shallowMount(EnterNames, { store, localVue })

        const expected = ["", ""]
        expect(component.vm.playerList).toEqual(expected)
      })
    })
  })

  describe("user interactions", () => {

    describe("events", () => {

      it("should call validateInputs when right arrow is clicked", () => {
        const spy = jest.spyOn(EnterNames.methods, "validateInputs")
        const component = shallowMount(EnterNames, { store, localVue })

        component.find("#right-arrow").trigger("click")

        expect(spy).toHaveBeenCalledTimes(1)
      })
    })
  })
})
