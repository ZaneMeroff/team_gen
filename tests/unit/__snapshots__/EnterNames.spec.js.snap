// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterNames snapshots should render error message if showErr is true 1`] = `
<div
  class="enter-names-container"
>
  <div
    class="nav-btn-container"
  >
    <router-link-stub
      ariacurrentvalue="page"
      class="router-link"
      event="click"
      tag="a"
      to="/"
    >
      <arrowbtn-stub
        arrowdirection="left"
      />
    </router-link-stub>
     
    <div
      id="right-arrow"
    >
      <arrowbtn-stub
        arrowdirection="right"
      />
    </div>
  </div>
   
  <div
    class="err-msg"
  >
    <!---->
  </div>
   
  <div
    class="name-inputs-container"
  >
    <nameinput-stub
      id="0"
      playername=""
    />
    <nameinput-stub
      id="1"
      playername=""
    />
  </div>
</div>
`;

exports[`EnterNames snapshots should render with NameInputs from playerList 1`] = `
<div
  class="enter-names-container"
>
  <div
    class="nav-btn-container"
  >
    <router-link-stub
      ariacurrentvalue="page"
      class="router-link"
      event="click"
      tag="a"
      to="/"
    >
      <arrowbtn-stub
        arrowdirection="left"
      />
    </router-link-stub>
     
    <div
      id="right-arrow"
    >
      <arrowbtn-stub
        arrowdirection="right"
      />
    </div>
  </div>
   
  <div
    class="err-msg"
  >
    <!---->
  </div>
   
  <div
    class="name-inputs-container"
  >
    <nameinput-stub
      id="0"
      playername="Ned"
    />
    <nameinput-stub
      id="1"
      playername="Ted"
    />
    <nameinput-stub
      id="2"
      playername="Sue"
    />
  </div>
</div>
`;

exports[`EnterNames snapshots should render with default html 1`] = `
<div
  class="enter-names-container"
>
  <div
    class="nav-btn-container"
  >
    <router-link-stub
      ariacurrentvalue="page"
      class="router-link"
      event="click"
      tag="a"
      to="/"
    >
      <arrowbtn-stub
        arrowdirection="left"
      />
    </router-link-stub>
     
    <div
      id="right-arrow"
    >
      <arrowbtn-stub
        arrowdirection="right"
      />
    </div>
  </div>
   
  <div
    class="err-msg"
  >
    <!---->
  </div>
   
  <div
    class="name-inputs-container"
  >
    <nameinput-stub
      id="0"
      playername=""
    />
    <nameinput-stub
      id="1"
      playername=""
    />
  </div>
</div>
`;
